mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -903,7 +903,7 @@ class LinearRegressionSummary private[regression] (
     val labelCol: String,
     val featuresCol: String,
     private val privateModel: LinearRegressionModel,
-    private val diagInvAtWA: Array[Double]) extends Serializable {
+    private val diagInvAtWA: Array[Double]) extends Summary with Serializable {
```

Comment:
out of curiosity, why?

Reply:
https://github.com/apache/spark/blob/2721a50234b05db3e59b5a37225f9358978a4590/sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/MLHandler.scala#L132-L145

The `FETCH` command handle `Summary` separately, we don't have a trait `RegressionSummary` (like `ClassificationSummary`), if we don't make `LinearRegressionSummary` extends `Summary` than the `FETCH` command will try to serialize the `LinearRegressionSummary` and cause failure